cargo-sqlx/src/main.rs

```diff
@@ -0,0 +1,206 @@
+use std::env;
+use std::fs;
+use std::fs::File;
+use std::io::prelude::*;
+
+use dotenv::dotenv;
+
+use sqlx::PgConnection;
+use sqlx::PgPool;
+
+use structopt::StructOpt;
+
+const MIGRATION_FOLDER: &'static str = "migrations";
+
+/// Sqlx commandline tool
+#[derive(StructOpt, Debug)]
+#[structopt(name = "Sqlx")]
+enum Opt {
+    // #[structopt(subcommand)]
+    Migrate(MigrationCommand),
+}
+
+/// Simple postgres migrator
+#[derive(StructOpt, Debug)]
+#[structopt(name = "Sqlx migrator")]
+enum MigrationCommand {
+    /// Initalizes new migration directory with db create script
+    // Init {
+    //     // #[structopt(long)]
+    //     database_name: String,
+    // },
+
+    /// Add new migration with name <timestamp>_<migration_name>.sql
+    Add {
+        // #[structopt(long)]
+        name: String,
+    },
+
+    /// Run all migrations
+    Run,
+}
+
+#[tokio::main]
+async fn main() {
+    let opt = Opt::from_args();
+
+    match opt {
+        Opt::Migrate(command) => match command {
+            // Opt::Init { database_name } => init_migrations(&database_name),
+            MigrationCommand::Add { name } => add_migration_file(&name),
+            MigrationCommand::Run => run_migrations().await,
+        },
+    }
+
+    println!("All done!");
+}
+
+// fn init_migrations(db_name: &str) {
+//     println!("Initing the migrations so hard! db: {:#?}", db_name);
+// }
+
+fn add_migration_file(name: &str) {
+    use chrono::prelude::*;
+    use std::path::Path;
+    use std::path::PathBuf;
+
+    if !Path::new(MIGRATION_FOLDER).exists() {
+        fs::create_dir(MIGRATION_FOLDER).expect("Failed to create 'migrations' dir")
+    }
+
+    let dt = Utc::now();
+    let mut file_name = dt.format("%Y-%m-%d_%H-%M-%S").to_string();
+    file_name.push_str("_");
+    file_name.push_str(name);
+    file_name.push_str(".sql");
+
+    let mut path = PathBuf::new();
+    path.push(MIGRATION_FOLDER);
+    path.push(&file_name);
+
+    if path.exists() {
+        eprintln!("Migration already exists!");
+        return;
+    }
+
+    let mut file = File::create(path).expect("Failed to create file");
+    file.write_all(b"-- Add migration script here")
+        .expect("Could not write to file");
+
+    println!("Created migration: '{}'", file_name);
+}
+
+pub struct Migration {
+    pub name: String,
+    pub sql: String,
+}
+
+fn load_migrations() -> Vec<Migration> {
+    let entries = fs::read_dir(&MIGRATION_FOLDER).expect("Could not find 'migrations' dir");
```

Comment:
Maybe a glob crate? https://crates.io/crates/glob

Reply:
I was thinking about it, but glob is a pretty big dependency that I would rather skip if it is not needed. If we start add more complicated rules I would be all for using a glob lib.

Reply:
That's fair. The dependency tree is already pretty big because of async.